ood-portal-generator/lib/ood_portal_generator/view.rb

```diff
@@ -22,6 +22,7 @@ def initialize(opts = {})
       @protocol         = @ssl ? "https://" : "http://"
       @listen_addr_port = opts.fetch(:listen_addr_port, nil)
       @servername       = opts.fetch(:servername, nil)
+      @serveraliases    = opts.fetch(:serveraliases, nil)
```

Comment:
```suggestion
      @server_aliases    = opts.fetch(:server_aliases, [])
```
If this defaults to an empty array do we need the if block in the template?  Also I do think I prefer the word separator.

Aside from that - one thing we have to consider is how to pass this list back to the dashboard. Rails needs to know these things -  https://guides.rubyonrails.org/configuring.html#actiondispatch-hostauthorization. 

I can't tell if this would be secure enough or if we'd actually need to pass this list back to the dashboard.
```
Rails.application.config.hosts = [
  IPAddr.new("0.0.0.0/0"),        # All IPv4 addresses.
  IPAddr.new("::/0"),             # All IPv6 addresses.
]
``

Reply:
I made the suggested change outside Github to also update template and example YAML and some tests.

As for configuring the hosts in Rails, I think the most reliable would be loading `/etc/ood/config/ood_portal.yml` with the initializer and pull out `servername` and `server_aliases` and possibly `proxy_server`.

Reply:
Because it can contain secrets, I think at least ansible writes ood_portal.yml as 600 and a spot check on our systems seems to imply the same for puppet.

We may be able to get them programmatically through an nslookup but I feel like that's the security flaw they were trying to avoid - [dns rebinding](https://en.wikipedia.org/wiki/DNS_rebinding)

Reply:
nslookup is pretty reliable but not sure how that would work.  You can pass FQDN/hostname or IPs to config.hosts. I checked and Satellite sets that value to FQDN of Satellite server (Foreman is upstream product).  To use nslookup you'd first have to know what hostname to lookup.  Is it not possible for Rails to get the hostname of the web service it's running behind?  Can we pass environment variables from Apache to Rails?  If Rails can't figure out the web address then maybe just passing it via environment variables if that's possible.

Reply:
If Rails can't figure out the address in initializer and if env vars can't be passed, the only other solution I can think of is having ood-portal-generator write out a world readable file that just lists hostnames OnDemand should respond to.

Reply:
We can likely use `Socket.ip_address_list` to get IPs and then look _them_ up, but reading the about attack (I'm not super familiar with it) it does require the attacker to have control of a DNS server.  Presumably it's one in the wild and not _ours_ (or a sites internal machine, for example OARNET's). So when _we_ (the dashboard) look up an IP we're likely communicating with a non-comprimised DNS server.

Though I don't actually know if that's a correct read of the attack vector. If the attacker would actually need to compromise OARNET's DNS for the attack to work - that I'm unclear on.

Reply:
If you just use `Socket.ip_address_list` to get local IPs, you can just push that to config.hosts as <IP>/32. The /32 designation is going to limit it to just that IP address.  The examples in docs use 0.0.0.0/0 so presume can use any valid CIDR notation such as 1.2.3.4/32 for specific IP.

In order for this attack to work at OSC, for example, the OSC DNS servers would need to be compromised.  It doesn't matter about OARnet or any other DNS server because OSC instructs all OSC systems to talk directly to our DNS servers. Any DNS lookup we can't serve, will go to OARnet or the internet but a reverse lookup on IP for our web servers will only hit OSC DNS servers.  This pattern is pretty common at most centers.  The smaller centers will end up using /etc/hosts or larger centers who might use tools that only know how to update /etc/hosts and push that out to all nodes.

Reply:
ENV variables could be passed - but then we're involving NGINX too, but honestly that may be the simplest (from the admin's perspective) & most secure way.

Reply:
OK - let's make this change, basically as is and with this link here #1530 can refer back to this conversation and we can solve it separately. I like the CIDR thing, but who knows if folks are using multiple IP addresses. passing ENV variables seems to be the best option in a list of bad options.